diff -rupN django_rules/backends.py django_rules.new/backends.py
--- django_rules/backends.py	2015-03-19 23:48:04.000000000 +1100
+++ django_rules.new/backends.py	2015-03-19 23:48:43.000000000 +1100
@@ -1,10 +1,10 @@
 # -*- coding: utf-8 -*-
 import inspect
+from importlib import import_module
 
 from django.conf import settings
 from django.contrib.contenttypes.models import ContentType
 from django.contrib.auth.models import User, AnonymousUser
-from django.utils.importlib import import_module
 
 from models import RulePermission
 from exceptions import NotBooleanPermission
